BUGFIX | Guide button - Update Help Page HidHide

OmarIskandarani-KSO opened the pull request:
I use / test this debug build locally, it is now able to work with the guide button. 
Tested with the "Thrustmaster T.Flight Hotas 4" 

![image](https://github.com/x360ce/x360ce/assets/111857216/6cd608ff-11ef-4b07-a727-86e101f77635)

fartcoin-project committed:
add guide button to work

OmarIskandarani-KSO committed:
Merge branch 'refs/heads/main' into feature/guide-button

# Conflicts:
#	x360ce.App.Beta/Common/DInput/DInputHelper.Step5.VirtualDevices.cs

OmarIskandarani-KSO committed:
broken

OmarIskandarani-KSO committed:
Merge branch 'master' into feature/guide-button

# Conflicts:
#	x360ce.App.Beta/Resources/x360ce.Engine.dll
#	x360ce.App.Beta/x360ce.App.csproj
#	x360ce.Engine/Data/PadSetting.cs
#	x360ce.Engine/packages.config
#	x360ce.Engine/x360ce.Engine.csproj
#	x360ce.RemoteController/x360ce.RemoteController.csproj

OmarIskandarani-KSO committed:
Merge branch 'master' into feature/guide-button

# Conflicts:
#	x360ce.Engine/JocysCom/Runtime/ChangeState.cs
#	x360ce.Engine/x360ce.Engine.csproj

AdrianTP commented:
@OmarIskandarani-KSO do you have build instructions for this pr so i can at least run it locally, even though it's been years and it still hasn't been merged to master?

OmarIskandarani-KSO committed:
Add project configuration files and update package references

OmarIskandarani-KSO commented:
I have used `Jetbrains Rider` (free community edition), and have the `full Visual Studio` Package installed (for .net / C#  windows apps dependencies),  open the project folder, choose the top project (x360CE.sln) 

Check:
In Settings -> Build, Execute, Deploy -> Toolset and Build

make sure `.NET CLI path` is correct
make sure MSBuild version is set to `C:\Program Files\Visual Sudio,,,`  (I have 17)

Then load the project and click the `red hammer` for build, then `green play` to deploy a release or debug..

I just migrated to a new windows, so I think you dont need much more,,   

You will probably get some dependency errors, just read them and solve..  

When you dont know copy paste the error message to chat-gpt for help.

Hope you can get it to work :D


### From README.md  (I did not have .NET2.0 and 3.0 from windows featured installed.)
---
[.NET 3.5  Programs and Features - Turn Windows Features on and off.](http://msdn.microsoft.com/en-us/library/hh506443.aspx)
[.NET 4.0 (link to 4.6, also installs 4.0)](http://www.microsoft.com/en-us/download/details.aspx?id=48130) (included in Windows 8 and 10)
.NET MUST be installed prior to the DirectX webupdate as it checks that .NET 2.0/3.0/3.5
[DirectX End-User Runtime (June 2010)](http://www.microsoft.com/en-us/download/details.aspx?id=8109) (required regardless of OS)
[Visual C++ Redistributable for Visual Studio 2013](http://www.microsoft.com/en-us/download/details.aspx?id=40784) (for x64 systems install both x86 and x64 redistributables)
Most game's executables are 32-bit and require x86 x360ce and also the x86 redistributable.

OmarIskandarani-KSO committed:
Add migration state configuration files for various components

Add project configuration files for SQL data sources and version control

Deprecate HidGuardian.
Now in favor of HidHide; update references and help content

OmarIskandarani-KSO committed:
Update help documentation to reflect changes from HIDGuardian to HIDHide

EJocys commented:
Thank you, @OmarIskandarani-KSO, and sorry this one sat for so long.

The guide button fix is in. It shipped in v4.17.46.0, credited to you in the changelog:

> - Fixed: Guide button was not passed to the virtual controller (PR #1473, OmarIskandarani-KSO).

Only that fix was taken. The branch also carried editor settings, a packaged zip and some unrelated project changes, and master has since been restored to the 4.17 baseline with a different project layout, so it could not be merged as it stood. The guide button change was reapplied by hand and credited to you.

So this is closed as applied, not as rejected. Thank you for chasing it down and for testing it on real hardware. That screenshot made it easy to see what you meant.

Thank You,
Evaldas Jocys
